### PR TITLE
Feature/54_notifications

### PR DESCRIPTION
**Pull request by khcr**
This feature let the user subscribe to notifications from its profile.
Its preferences and token are stored in the database for further usages.

### CODE REVIEW

File: app/src/main/java/com/ncnf/notification/Registration.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import com.google.firebase.messaging.FirebaseMessaging;
+import com.ncnf.user.PrivateUser;
+
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+
+public class Registration {
+
+    private FirebaseMessaging service;
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/notification/Registration.java:12]: Make attributes final?

File: app/src/main/java/com/ncnf/user/UserProfileActivity.java [hunk 2]
```diff
@@ -23,24 +28,33 @@
...
+
     private boolean firstNameChanged = false;
     private boolean lastNameChanged = false;
     private boolean birthDateChanged = false;
+
     private EditText email;
     private EditText firstName;
     private EditText lastName;
     private EditText birthDate;
+    Switch notification_switch;
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/user/UserProfileActivity.java:55]: private?

**Review by HyPn0TiiK — APPROVED** [review]

File: app/src/main/java/com/ncnf/notification/Registration.java (hunk 1 above)
Comment (khcr) [reply app/src/main/java/com/ncnf/notification/Registration.java]: Correct!

File: app/src/main/java/com/ncnf/user/UserProfileActivity.java (hunk 2 above)
Comment (khcr) [reply app/src/main/java/com/ncnf/user/UserProfileActivity.java]: Correct!

**Pull request merged**